ansible_runner/cleanup.py

```diff
@@ -0,0 +1,192 @@
+             "This is to avoid deleting folders that were recently created, or folders not started via the start command. "
+             "Value of 0 indicates that no folders will be excluded based on modified time."
+    )
+    command.add_argument(
+        "--image-prune",
+        action="store_true",
+        help="If specified, will run docker / podman image prune --force. "
+             "This will only run after untagging."
+    )
+    command.add_argument(
+        "--process-isolation-executable",
+        default="podman",
+        help="The container image to clean up images for (default=podman)"
+    )
+
+
+def run_command(cmd):
+    '''Given list cmd, runs command and returns standard out, expecting success'''
+    process = subprocess.run(cmd, capture_output=True)
+    stdout = str(process.stdout, encoding='utf-8')
+    if process.returncode != 0:
+        print('Error running command:')
+        print(' '.join(cmd))
+        print('Stdout:')
+        print(stdout)
+        raise RuntimeError('Error running command')
+    return stdout.strip()
+
+
+def is_alive(dir):
+    pidfile = os.path.join(dir, 'pid')
+
+    try:
+        with open(pidfile, 'r') as f:
+            pid = int(f.readline())
+    except IOError:
+        return False
+
+    try:
+        os.kill(pid, signal.SIG_DFL)
+        return(0)
+    except OSError:
+        return(1)
+
+
+def project_idents(dir):
+    """Give dir, give list of idents that we have artifacts for"""
+    try:
+        return os.listdir(os.path.join(dir, 'artifacts'))
+    except (FileNotFoundError, NotADirectoryError):
+        return []
+
+
+def delete_associated_folders(dir):
+    """Where dir is the private_data_dir for a completed job, this deletes related tmp folders it used"""
+    for ident in project_idents(dir):
+        registry_auth_pattern = f'/tmp/{registry_auth_prefix}{ident}_*'
+        for dir in glob.glob(registry_auth_pattern):
+            changed = cleanup_folder(dir)
+            if changed:
+                print(f'Removed associated registry auth dir {dir}')
+
+
+def validate_pattern(pattern):
+    # do not let user shoot themselves in foot by deleting these important linux folders
+    prohibited_paths = set(Path(s) for s in (
+        '/', '/bin', '/dev', '/home', '/lib', '/mnt', '/proc',
+        '/run', '/sys', '/usr', '/boot', '/etc', '/opt', '/sbin', '/tmp', '/var'
+    ))
+    bad_paths = [dir for dir in glob.glob(pattern) if Path(dir).resolve() in prohibited_paths]
+    if bad_paths:
+        raise RuntimeError(
+            f'Provided pattern could result in deleting system folders:\n{" ".join(bad_paths)}\n'
+            'Refusing to continue for user system safety.'
+        )
+
+
+def cleanup_dirs(pattern, exclude_strings=(), grace_period=GRACE_PERIOD_DEFAULT):
+    try:
+        validate_pattern(pattern)
+    except RuntimeError as e:
+        print(str(e))
+        sys.exit(1)
```

Comment:
Passing a string to `sys.exit()` exits 1 printing the string to `stderr`. That's slightly different than the current code here but printing to `stderr` is probably more accurate.
```suggestion
        sys.exit(str(e))
```